src/contexts/RemoteConfig/index.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react'
+import PropTypes from 'prop-types'
+import { useI18n } from '../I18n'
+
+type RemoteConfig = {
+  /**
+   * An object containing the remotely fetched key / value pairs.
+   * @note Locally, defaults to data exported from `src/api/cms/settings.js`
+   *
+   * @note RECOMMENDATION: When a config is set up in a CMS and passed
+   * into this codebase, please remove the contents of this type and replace
+   * it with the actual type data from the CMS.
+   */
+  [key in string]: unknown
+}
+
+type RemoteConfigCTX = RemoteConfig | undefined
+export const RemoteConfigContext = React.createContext<RemoteConfigCTX>(undefined)
+
+if (process.env.NODE_ENV !== 'production') {
+  RemoteConfigContext.displayName = 'RemoteConfigContext'
+}
+
+export function useRemoteConfig() {
+  const context = React.useContext(RemoteConfigContext)
+  if (!context) {
+    throw new Error('useRemoteConfig must be used within a RemoteConfigProvider')
+  }
+  return context
+}
+
+interface RemoteConfigProviderProps {
+  children: React.ReactNode
+  value?: RemoteConfig
+}
+
+function RemoteConfigProvider(props: RemoteConfigProviderProps) {
+  const { children, value = {} } = props
+
+  const { locale } = useI18n()
```

Comment:
Looks great @Dashice!

Just noticed this while parsing the code and I realise it's already part of the codebase but is there a specific reason locale isn't passed a prop here @maeertin? 

Currently it implicitly forces a specific [context order](https://github.com/noaignite/create-ignite-app/blob/main/src/contexts/Root/index.js#L28) which I feel this component shouldn't be responsible for, but maybe I'm missing something?

Reply:
Very true @janmysiak, we might as well pass it as a prop. Should that be part of this PR?